sdk/provisioning/Azure.Provisioning.CloudMachine/src/AzureSdkExtensions/OpenAIFeature.cs

```diff
@@ -8,21 +8,30 @@
 using Azure.Provisioning.Authorization;
 using Azure.Provisioning.CognitiveServices;
 using OpenAI.Chat;
+using OpenAI.Embeddings;
 
 namespace Azure.Provisioning.CloudMachine.OpenAI;
 
 public class OpenAIFeature : CloudMachineFeature
 {
-    public string Model { get; }
-    public string ModelVersion { get; }
+    private OpenAiModelDeployment _chatDeployment;
+    private OpenAiModelDeployment? _embeddingsDeployment;
 
-    public OpenAIFeature(string model, string modelVersion) {  Model = model; ModelVersion = modelVersion; }
+    public OpenAIFeature(OpenAiModelDeployment chatDeployment, OpenAiModelDeployment? embeddingsDeployment = default)
```

Comment:
ultimately we need to support a collection of deployments, and so I am not thrilled with special casing for 2 deployments.

Reply:
I started down that path, but there are some problems to be solved with how the models are represented in `GetConnectionOptions`. This approach allows us to retain some reasonable defaults. I think we can refactor this as the need arises for more deployed model types.

Reply:
I think it's fine to do this as a separate PR.